src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -121,17 +118,17 @@ export const getModuleDetails = async (
   );
 
   // TODO: Customize this block lookback based on chain and test with L2 network (Polygon)
-  const events = await oracleContract.queryFilter('ProposePrice', -100000);
-  const proposeEvent = events.filter(
+  const proposalEvents = await oracleContract.queryFilter('ProposePrice');
+  const thisModuleProposalEvents = proposalEvents.filter(
```

Comment:
I think requester is an indexed parameter on ProposePrice. Prefer adding the requester filter to the queryFilter so the node will return fewer results (and you will be less likely to hit some sort of node return size limit).

Reply:
Interesting, I'll take a look at how to implement that and make the change.

Reply:
I don't see a way to add more details than the event name to `queryFilter`. Digging around in the docs to see if there is another path, but do you know how I could do pre-filtering? https://docs.ethers.org/v5/api/contract/contract/#Contract-queryFilter

Reply:
I do see that only indexed event parameters can be filtered, which is why we can filter by `requestor` in the next function. https://github.com/UMAprotocol/snapshot/pull/25#discussion_r1061779431